backend/main.py

```
@@ -118,3 +118,49 @@ def read_sales_by_year(year: int, session: SessionDep):
+
+# Word routes
+@app.post("/words/", response_model=schemas.Word)
+def create_word(word: schemas.WordCreate, session: SessionDep):
+    db_word = crud.get_word_by_text(session, word.word)
+    if db_word:
+        raise HTTPException(status_code=400, detail="Word already exists")
+    try:
+        return crud.create_word(session, word)
+    except Exception as e:
+        session.rollback()
+        raise HTTPException(status_code=500, detail=f"Failed to create word: {str(e)}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)